### ISSUE TITLE
Issue with document generation (quotes/invoices), margin and pagination

### ISSUE THREAD

**Issue by ipdeveloppement44** [Dolibarr/dolibarr#28274]
### Bug

On document templates, specifically the one for invoices labeled "crabe," the price of certain lines is written in the footer. The second issue is that the pagination is also incorrect. For a 3-page document, the first page shows "1/2," the second page "2/3," and the last page "3/3."

### Environment Version

18.0.3

### Environment OS

Linux

### Environment Web server

Apache

### Environment PHP

8.1

### Environment Database

MariaDB 10.3.39

### Expected and actual behavior

The PDFs are configured to display the price at the top of the line (MAIN_PDF_DATA_ON_FIRST_PAGE = 1). Therefore, if a line starts slightly at the bottom of the previous page, it writes the prices at that level. There is no issue with the description, but in terms of writing prices, they end up in the footer.

The second problem concerns pagination, where for a 3-page document, the first page displays "1/2," the second page "2/3," and the last page "3/3."

### Attached files

Issue with writing the price:
![pdf1](https://github.com/Dolibarr/dolibarr/assets/132068748/38437f3a-f769-4864-847e-4b4e8011847e)
Pagination issue:
![pdf2](https://github.com/Dolibarr/dolibarr/assets/132068748/8864db3f-f94b-4bc4-b23d-7908d96bb105)
![pdf3](https://github.com/Dolibarr/dolibarr/assets/132068748/71c4fb08-b50b-493d-9f12-67a5e62b6518)
![pdf4](https://github.com/Dolibarr/dolibarr/assets/132068748/4eb46e06-ce4d-48a6-8155-9d687ab00f72)
![pdf5](https://github.com/Dolibarr/dolibarr/assets/132068748/d86af52e-3491-44a7-9bb4-de5d68dfc5ba)

**Comment by badrbouslikhin** [comment]
I'm experiencing the same issue with the `cyan` propale template on v19.0.0.
The issue comes from [this line](https://github.com/Dolibarr/dolibarr/blob/8ca94fd5217a093586c7f932fd436bcef903cfee/htdocs/core/modules/propale/doc/pdf_cyan.modules.php#L642). The `empty($showpricebeforepagebreak)` condition leads to not updating the page on pagebreak.
I am still looking for a fix that preserves both a proper formatting and `MAIN_PDF_DATA_ON_FIRST_PAGE = 1` option.

**Comment by badrbouslikhin** [comment]
Here is a dirty hack I found (knowing that my propale template is derived from cyan):
```diff
diff --git a/dolibarr/modeles/propale/pdf_thermigo.modules.php b/dolibarr/modeles/propale/pdf_thermigo.modules.php
index 4ab9c8e..ca9ec4d 100644
--- a/dolibarr/modeles/propale/pdf_thermigo.modules.php
+++ b/dolibarr/modeles/propale/pdf_thermigo.modules.php
@@ -607,6 +607,11 @@ class pdf_thermigo extends ModelePDFPropales
 
               $this->printColDescContent($pdf, $curY, 'desc', $object, $i, $outputlangs, $hideref, $hidedesc);
 
+              // HACK: to handle the case where there is a pagebreak before the description
+              // and the price is written on the previous page.
+              $curYAfterPageBreakWrite = $curY;
+              $getYAfterPageBreakWrite = $pdf->GetY();
+
               $pageposafter = $pdf->getPage();
               $posyafter = $pdf->GetY();
               //var_dump($posyafter); var_dump(($this->page_hauteur - ($heightforfooter+$heightforfreetext+$heightforinfotot))); exit;
@@ -646,6 +651,13 @@ class pdf_thermigo extends ModelePDFPropales
             $curY = $tab_top_newpage;
           }
 
+          // HACK: to handle the case where there is a pagebreak before the description
+          // and the price is written on the previous page.
+          if ($pageposafter > $pageposbefore && $getYAfterPageBreakWrite < $curYAfterPageBreakWrite) {
+            $pdf->setPage($pageposafter);
+            $curY = $tab_top_newpage;
+          }
+
           $pdf->SetFont('', '', $default_font_size - 1); // We reposition the default font
 
           // VAT Rate
```

**Comment by fappels** [comment]
Progress on fix here: Issue #29458

**Comment by badrbouslikhin** [comment]
@fappels, I tried the fix that I believe was included in v19.0.2, and the pagination issue is indeed resolved. However, the price of certain lines written in the footer as described by @ipdeveloppement44 is still present for me.

My [workaround](https://github.com/Dolibarr/dolibarr/issues/28274#issuecomment-2076836047) addresses the issue for proposals with short description items, but it causes the price on the first page to break for items with long descriptions that span multiple pages.

Do you have any suggestions on where I should investigate further to resolve this issue?

**Comment by fappels** [comment]
Hello @badrbouslikhin , did not notice there were two issue in this report. 
Seems second issue coming from HIDDEN option introduced here: #13005 
Think your workaround ignores this option. Needs some debugging for fixing the HIDDEN option so can give no suggestion on this.
Think @ipdeveloppement44 should make new issue on HIDDEN option  MAIN_PDF_DATA_ON_FIRST_PAGE.
